src/ert/scheduler/scheduler.py

```diff
@@ -215,14 +225,18 @@ async def _monitor_and_handle_tasks(
                     )
                     logger.error(
                         (
-                            f"Exception in scheduler task: {task_exception}\n"
+                            f"Exception in scheduler task {task.get_name()}: {task_exception}\n"
```

Comment:
This should perhaps be a separate commit, as it is unrelated to fixing the premature closing of publisher.

Reply:
Now it's in two commits. 1) improving exception messages 2) the actual fix